src/core/thread/meshcop_tlvs.hpp

```diff
@@ -659,6 +660,59 @@ class ActiveTimestampTlv : public Tlv, public Timestamp
 } OT_TOOL_PACKED_END;
 
 /**
+ * This class implements State TLV generation and parsing.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class StateTlv: public Tlv
```

Comment:
You should borrow the implementation from my [pr/commissioning](https://github.com/jwhui/openthread/blob/pr/commissioning/src/core/thread/meshcop_tlvs.hpp#L936) branch.  See further comments below.